pallets/providers/README.md

```diff
@@ -378,3 +377,45 @@ Error thrown when trying to get a root from a Main Storage Provider without pass
 ### `SpRegisteredButDataNotFound`
 
 Error thrown when a user has a Storage Provider ID assigned to it but its metadata data does not exist in storage (storage inconsistency error, should never happen).
+
+## Slashing Protocol
+
+Storage Providers who fail to submit a proof by the last challenge tick will be slashed, predetermined by the challenge period defined in the proofs-dealer pallet.
+
+Slashing is an asynchronous process, therefore it is possible for a Storage Provider to have failed more than one challenge before being slashed. To avoid all possibility for a Storage Provider to not be slashed for the number of failed proof submissions, the runtime will accrue the number of failed challenges for each Storage Provider. Slashing a Storage Provider will take into account the total number of failed challenges and multiply it by a configurable slash factor.
+
+### Manual and Automatic Slashing
+
+The `slash` extrinsic can be called by any account to manually slash a Storage Provider and only requires the Storage Provider ID of a Storage Provider to be slashed, be it either an MSP or a BSP.
+
+An automated slashing mechanism is implemented in an off-chain worker process to be executed by collators which efficiently slashes many Storage Providers.
+
+### Grace Period and Insolvency
+
+Since the Storage Provider's stake determines their total storage capacity, it is entirely possible that the amount of data currently stored would be above their total storage capacity. StorageHub grants a predetermined configurable grace period for Storage Providers to top-up their stake to have their total capacity equal to or greater than the amount of data they currently store. If the grace period has been reached, they are considered to be insolvent and cease to be a Storage Provider.
```

Comment:
```suggestion
Since the Storage Provider's stake determines their total storage capacity, it is entirely possible that the amount of data currently stored would be above their total storage capacity after slashing. StorageHub grants a predetermined configurable grace period for Storage Providers to top-up their stake to have their total capacity equal to or greater than the amount of data they currently store. If the grace period has been reached, they are considered to be insolvent and cease to be a Storage Provider.
```